### DISCUSSION TITLE
Welcome to crank-organ Discussions!

### DISCUSSION THREAD

**Discussion by bixb922** [bixb922/crank-organ#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

 Also: Please post here if you are using or want to use the software, and how you are using it (i.e are you using microphone? crank sensor? etc).

Here is my crank organ channel on Youtube: https://www.youtube.com/@bixb922

**Comment by joholes** [comment]
Dear friends,
 I have successfully installed the crank-organ software. I want to use it as a replacement for the destroyed cylinder of a 100-year-old barrelorgan (46 pipe, 37 valves - now solenoid). I control the solenoids with the Arduino MEGA via the Rx pin from an external MIDI player. SW bixb is ingenious and very variable. Now I have esp32s3 with 2 mcp23017+GPIO OK in testing. For playing at home, I want to add the possibility of playing simultaneously over a MIDI serial line to an external midi instrument to complement the playing of a midi file from another channel. From the SW/data/64_note_midi.. variant. I can play MIDI data. In the doc-SW (Select scale) text, there is a mention of the possibility of combinations. However, I got lost here (I'm 70+). Please, can you help me with the configuration suggestion: 2 mcp 23017+5GPIO (=37 solenoids) +MIDI Tx serial - of course, if I don't bother you too much. At the same time, I apologize for the need to use Google Translate. Thank you for a beautiful job. Joholes, Czech Republic

**Reply by bixb922** [reply]
Hello  @joholes,
Thank you for your post.  I have started a new discussion with this topic. Please see my answer here #12 in this same forum.
